README.md

```diff
@@ -1,5 +1,117 @@
+[![Build Status](http://img.shields.io/travis/metrics-js/emitter/master.svg?style=flat-square)](https://travis-ci.org/metrics-js/emitter)
+[![Greenkeeper badge](https://badges.greenkeeper.io/metrics-js/emitter.svg?style=flat-square)](https://greenkeeper.io/)
+[![Known Vulnerabilities](https://snyk.io/test/github/metrics-js/emitter/badge.svg?targetFile=package.json&style=flat-square)](https://snyk.io/test/github/metrics-js/emitter?targetFile=package.json)
+
+## Installation
+
+```bash
+$ npm install @metrics/emitter
+```
+
+## Example
+
+Set up an emitter which connects to a daemon on UDP on port 6000 and  pipes the metrics from the [`@metrics/client`](https://github.com/metrics-js/client):
+
+```js
+const Emitter = require('@metrics/emitter');
+const Client = require('@metrics/client');
+
+const emitter = new Emitter('udp', { port: 6000 });
+const client = new Client();
+
+client.pipe(emitter);
+```
+
+## Description
+
+This module makes it possible stream metrics over a socket to a daemon. The socket can
+be of different protocols (UDP, TCP etc) but the data trasmitted over it is expected to be of
+the [`@metrics/metric`](https://github.com/metrics-js/metric) type.
+
+For recieving metrics over a socket one are expected to use the [`@metrics/daemon`](https://github.com/metrics-js/daemon) module.
+
+The main purpose of this is to be able to send metrics from multiple processes to a sentral
+service (daemon) for further handling. Here is a simplified example of each worker in a
+cluster pushing metrics to the master and the master pushing the metric further:
+
+```js
+const master = () => {
+    const daemon = new Daemon('udp', { port: 6000 });
+    const client = new Client();
+
+    daemon.pipe(client);
+
+    daemon.listen();
+};
+
+const worker = () => {
+    const emitter = new Emitter('udp', { port: 6000 });
+    const client = new Client();
+
+    client.pipe(emitter);
+
+    client.metric({
+        name: `worker_${cluster.worker.id}`,
+        description: `Worker number: ${cluster.worker.id}`,
+        value: 1,
+    });
+};
+
+const workers = [];
+
+if (cluster.isMaster) {
+    for (let i = 0; i < (os.cpus().length - 1); i++) {
+        workers.push(cluster.fork());
+    }
+    master();
+}
+
+if (cluster.isWorker) {
+    worker();
+}
+```
+
+See the [cluster example](https://github.com/metrics-js/daemon/tree/master/example/cluster.js)
+in daemon for a full example.
+
+## Constructor
+
+Create a new Emitter instance.
+
+```js
+const Emitter = require('@metrics/emitter');
+const emitter = new Emitter(transport, options);
+```
+
+### transport (required)
+
+What type of transport to use. Supported values are:
+
+ * `udp` - For UDP transport.
+
+### options (optional)
+
+An Object containing misc configuration for the selected transport. Please see each
+transport for which option which can be passed in.
+
+### returns
+
+Returns a Writable stream in object mode.
+
+## Transports
```

Comment:
does it only support 1 transport for now, more to come?

Reply:
Currently only one to get us going. Will add TCP and probably WebSocket support also.